### PR TITLE
1.3.0-0.4.0: Feature - accounting exports, side nav & menu updates

### PR DESCRIPTION
**Pull request by johngribbin**
- Adds /bkpr route with 4 CSV download options
Screen Shot 2023-01-31 at 10 08 26 AM

- Adds "SideNav" component to homepage on Desktop devices (and relocated "Refresh" component)
Screen Shot 2023-02-01 at 11 35 18 AM

- Updates "Menu" component to become burger menu on homepage (mobile devices)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "browser-app",
-  "version": "1.3.0",
+  "version": "1.3.0-0.3.0",
```
Comment (johngribbin) [inline package.json:3]: `enhancement/connection-options` should be merged first

File: package.json [hunk 2]
```diff
@@ -49,6 +49,7 @@
     "big.js": "^6.2.1",
     "crypto-js": "^4.1.1",
     "date-fns": "^2.28.0",
+    "date-fns-tz": "^1.3.7",
```
Comment (johngribbin) [inline package.json:52]: Required to set UTC time for accounting exports

**Comment by aaronbarnardsound** [comment]
I created a branch [here](https://github.com/clams-tech/browser-app/tree/feature/bookkeeper-account-ui-tweaks) to experiment with some different UI patterns for the nav on desktop and mobile.

I think that we no longer need to use the node "color" for the dot, as it is not really relevant, instead it is now a WebSocket connection status indicator so you know if you are currently connected, connecting, disconnected.

This is also an attempt a nav without burger on mobile. I think the icons are intuitive enough by themselves, and worst case is a new user presses them to find out what they do. Once they have done that once for each button, then they know what they do and we have maintained a sleek, minimal interface. I figure on mobile we won't have that many more routes, so there is plenty of room up the top and if needed, I think we could also remove the payments and accounting route icons down the line to make room for more important routes on mobile.

**Comment by aaronbarnardsound** [comment]
Looking really good! I validated that the CSV's matched the `bkpr-dumpincomecsv` output and they match pretty much perfectly. The only thing I did notice is that the csv from bkpr always had 10 decimal places (msats precision) where as our output did not have the trailing zeros. I wouldn't think that would be an issue though.

I did have a go at a small refactor at some of the formatting logic on [this branch](https://github.com/clams-tech/browser-app/tree/refactor/csv-formatting) if you would like to take a look and consider merging in. What I wanted to achieve with the refactor:

- Minimise computation. I imagine some nodes will have huuuuge amounts of data, so we want to do the minimal computation needed and only when a download button is clicked. So I removed all of the reactive (`$:`) logic and moved it in to discrete functions, whilst attempting to simplify the logic to make it more readable as to what was happening and why.
- Simplified the dynamic import logic. Once a module is dynamically imported once, it is cached, so it is fine to put the import at the top of each format function.
- Added a loading spinner state to the buttons since the dynamic import may take a while on slow connections and the formatting may also take a while on slower devices that have a lot of data to process.

**Review by aaronbarnardsound — APPROVED** [review]
LFG!

**Pull request merged**